### PR TITLE
fix: update server setup script for DO

### PR DESCRIPTION
**Pull request by parketh**
## Summary

This PR updates the server setup Ansible script to work with DigitalOcean. 

## Test plan

```
ANSIBLE_HOST_KEY_CHECKING=False ansible-playbook -i ini/server.ini playbooks/debian_server_setup_playbook.yml
```

### CODE REVIEW

**Comment by lesterli** [comment]
after testing with `root`, it throws the error:

```
TASK [Install Oh My Zsh] ***************************************************************************************************************************************************************************************************************************************************************
fatal: [167.71.165.11]: FAILED! => {"changed": true, "cmd": "sudo -u snapchain sh -c \"$(curl -fsSL https://raw.github.com/ohmyzsh/ohmyzsh/master/tools/install.sh)\" \"\" --unattended", "delta": "0:00:00.709604", "end": "2024-10-30 03:37:11.929939", "msg": "non-zero return code", "rc": 2, "start": "2024-10-30 03:37:11.220335", "stderr": "From https://github.com/ohmyzsh/ohmyzsh\n * [new branch]      master     -> origin/master\nAlready on 'master'\n: 325: cd: can't cd to /root", "stderr_lines": ["From https://github.com/ohmyzsh/ohmyzsh", " * [new branch]      master     -> origin/master", "Already on 'master'", ": 325: cd: can't cd to /root"], "stdout": "Cloning Oh My Zsh...\nbranch 'master' set up to track 'origin/master'.", "stdout_lines": ["Cloning Oh My Zsh...", "branch 'master' set up to track 'origin/master'."]}

PLAY RECAP *****************************************************************************************************************************************************************************************************************************************************************************
167.71.165.11              : ok=6    changed=4    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0  
```

and fix the error by modifying the `Install Oh My Zsh` task, adding `HOME=/home/{{ shared_user }} `to cmd
```
    - name: Install Oh My Zsh
      ansible.builtin.shell:
        cmd: |
          HOME=/home/{{ shared_user }}
          sudo -u {{ shared_user }} sh -c "$(curl -fsSL https://raw.github.com/ohmyzsh/ohmyzsh/master/tools/install.sh)" "" --unattended
        creates: "/home/{{ shared_user }}/.oh-my-zsh"
```

File: README.md [hunk 1]
```diff
@@ -72,6 +72,8 @@ To just configure the server, run:
 ANSIBLE_HOST_KEY_CHECKING=False ansible-playbook -i ini/server.ini playbooks/debian_server_setup_playbook.yml
 ```
 
+For DigitalOcean servers, use `root` user to setup the server.
```
Comment (bap2pecs) [inline README.md:75]: we should also fix all ini example files

**Review by bap2pecs — COMMENTED** [review]
lgtm

File: README.md (hunk 1 above)
Comment (bap2pecs) [reply README.md:75]: https://github.com/Snapchain/devops/pull/27

**Pull request merged**